lib/bowling_scorecard.rb

```diff
@@ -0,0 +1,179 @@
+require_relative 'frame'
+
+class BowlingScorecard
+  attr_reader :current_score, :frame
+
+  def initialize
+    @frame = 1
+    @first_roll = nil
+    @second_roll = nil
+    @roll_score = nil
+    @strike = false
+    @spare = false
+    @bonus_points = 0
+    @frame_score = 0
+    @current_score = 0
+    @score_log = []
+  end
+
+  def enter_roll(score)
+    invalid_message = "Invalid score entered, score must be between 0 and #{max_score}."
+    return invalid_message unless valid?(score)
+
+    score = score.to_i
+    roll_turn(score)
+    end_game_condition ? "End Game" : score
+  end
+
+  def generate_scorecard_info
+    @score_log
+  end
+
+private
```

Comment:
so many private methods. Did you first have them within the public ones and then separated?